### PR TITLE
Add variant: NUCLEO-H503RB.

### PR DESCRIPTION
**Pull request by dojyorin**
### Summary
This PR fixes/implements the following **bugs/features**

- Add new variant [`NUCLEO-H503RB`](https://www.st.com/ja/evaluation-tools/nucleo-h503rb.html)

@fpistm feat adds [STM32H5](https://github.com/stm32duino/Arduino_Core_STM32/issues/1978) core implementation and NUCLEO-H563ZI support (That was really great help! Thank you!), but variant support is still lacking.

This time, I'll add new H5 NUCLEO board NUCLEO-H503RB with this pull request.
Related to https://github.com/stm32duino/Arduino_Core_STM32/issues/722#issuecomment-1594863252 and https://github.com/stm32duino/Arduino_Core_STM32/pull/2091

### Progress
There's still some work to do.

- [x] Adapt [variant_NUCLEO_H503RB.cpp](https://github.com/dojyorin/Arduino_Core_STM32/blob/936a8cc34cfb0f4fe50aab90b020f264143ad993/variants/STM32H5xx/H503RBT/variant_NUCLEO_H503RB.cpp#L18) and [variant_NUCLEO_H503RB.h](https://github.com/dojyorin/Arduino_Core_STM32/blob/936a8cc34cfb0f4fe50aab90b020f264143ad993/variants/STM32H5xx/H503RBT/variant_NUCLEO_H503RB.h#L18) pin assignments to actual board values.
- [x] Regarding [peripheral clock](https://github.com/dojyorin/Arduino_Core_STM32/blob/936a8cc34cfb0f4fe50aab90b020f264143ad993/variants/STM32H5xx/H503RBT/variant_NUCLEO_H503RB.cpp#L161) supply, there are no definitions for commonly used peripherals such as I2C or SPI, is it already defined elsewhere?
- [x] There any other missing parts?

Basically implementation of NUCLEO-H563ZI is used reference.

I'm new to STM32, so excuse me if I'm doing something wrong.
I would like to discuss with everyone and produce good results.

### CODE REVIEW

**Comment by fpistm** [comment]
Thanks for squashing.  Anyway simply force update on the same branch will avoid creating new PR. :wink : Let ci run to check this PR first.

File: variants/STM32H5xx/H503RBT/variant_NUCLEO_H503RB.h [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+#define PC8_ALT1                (PC8  | ALT1)
+#define PC9_ALT1                (PC9  | ALT1)
+#define PC12_ALT1               (PC12 | ALT1)
+
+#define NUM_DIGITAL_PINS        49
+#define NUM_ANALOG_INPUTS       16
+
+// On-board LED pin number
+#ifndef LED_BUILTIN
+  #define LED_BUILTIN           PNUM_NOT_DEFINED
```
Comment (fpistm) [inline variants/STM32H5xx/H503RBT/variant_NUCLEO_H503RB.h:118]: Should be defined to the pin number of the LED

File: variants/STM32H5xx/H503RBT/variant_NUCLEO_H503RB.h [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+#endif
+
+// On-board user button
+#ifndef USER_BTN
+  #define USER_BTN              PNUM_NOT_DEFINED
```
Comment (fpistm) [inline variants/STM32H5xx/H503RBT/variant_NUCLEO_H503RB.h:123]: Should be defined to the pin number of the user button (blue button)

File: variants/STM32H5xx/H503RBT/variant_NUCLEO_H503RB.h [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+  #define PIN_SPI_SCK           PA1
+#endif
+
+// I2C definitions
+#ifndef PIN_WIRE_SDA
+  #define PIN_WIRE_SDA          PB3
+#endif
+#ifndef PIN_WIRE_SCL
+  #define PIN_WIRE_SCL          PB3
+#endif
```
Comment (fpistm) [inline variants/STM32H5xx/H503RBT/variant_NUCLEO_H503RB.h:155]: When pin properly ordered no need to define this as by default they are mapped to D10-D13 (SPI) and D14/D15 (I2C)

File: variants/STM32H5xx/H503RBT/variant_NUCLEO_H503RB.h [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+
+// Default pin used for generic 'Serial' instance
+// Mandatory for Firmata
+#ifndef PIN_SERIAL_RX
+  #define PIN_SERIAL_RX         PA1
+#endif
+#ifndef PIN_SERIAL_TX
+  #define PIN_SERIAL_TX         PA2
+#endif
```
Comment (fpistm) [inline variants/STM32H5xx/H503RBT/variant_NUCLEO_H503RB.h:178]: Did you check the STLink VCP are connected to this U(S)ART pins

File: variants/STM32H5xx/H503RBT/generic_clock.c [hunk 5]
```diff
@@ -10,18 +10,85 @@
...
+*   SOURCE / PLLM * PLLN
+*
+* <TARGRT>
+*   PLL ... MULTIPLEXER / PLLP
+*
+* - Default CPU clock: 4(CSI) / 2(PLLM) * 250(PLLN) / 2(PLLP) ... 250 MHz
+* - PLL3 supports only H562/H563/H573.
+*/
+
+WEAK void SystemClock_Config(void) {
```
Comment (fpistm) [inline variants/STM32H5xx/H503RBT/generic_clock.c:35]: Astyle fix
```suggestion
WEAK void SystemClock_Config(void)
{
```

File: variants/STM32H5xx/H503RBT/variant_NUCLEO_H503RB.cpp [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+*   SOURCE / PLLM * PLLN
+*
+* <TARGRT>
+*   PLL ... MULTIPLEXER / PLLP
+*
+* - Default CPU clock: 24(HSE) / 12(PLLM) * 250(PLLN) / 2(PLLP) ... 250 MHz
+* - PLL3 supports only H562/H563/H573.
+*/
+
+WEAK void SystemClock_Config(void) {
```
Comment (fpistm) [inline variants/STM32H5xx/H503RBT/variant_NUCLEO_H503RB.cpp:116]: Astyle fix
```suggestion
WEAK void SystemClock_Config(void)
{
```

**Review by fpistm — CHANGES_REQUESTED** [review]
First thanks for this PR.
In addition to my comments, could you add the generic and the Nucleo in the README.md and also declare them to the boards.txt.
Finally, split in 2 commit, 1 for the generic addition and one for the dedicated Nucleo board.

You can refers to this PR as example:
https://github.com/stm32duino/Arduino_Core_STM32/pull/1859

**Comment by dojyorin** [comment]
OK, I understand the coding conventions.
I'll try to work on it over the weekend.

**Comment by dojyorin** [comment]
Thank you for detailed explanation, it really helped me.
It was a great reference when I wanted to add other boards in the future.

For the time being, I've fixed the part you told me, so I'll commit it.

**Comment by dojyorin** [comment]
The Arduino pinout only goes up to `D15`/`A5`, but the MCU has more pins.
I wasn't sure what rules to define the "surplus `D16`/`A6` and beyond" pins.

**Comment by fpistm** [comment]
Hi @dojyorin 
I've reworked and fixed your PR.
Could you try it and force update to your branch. Thanks
